dali/kernels/common/split_shape.h

```diff
@@ -33,17 +33,21 @@ namespace kernels {
  * @param in_shape Input shape
  * @param min_nblocks Desired minimum number of blocks
  * @param min_sz Minimum practical block size
+ * @param skip_dim If provided, this dimension will not be split
+ * @return product of split_factor
  */
 template <typename SplitFactor, typename Shape>
-void split_shape(SplitFactor& split_factor, const Shape& in_shape, int min_nblocks = 8,
-                 int min_sz = (16 << 10)) {
+int split_shape(SplitFactor& split_factor, const Shape& in_shape, int min_nblocks = 8,
+                int min_sz = (16 << 10), int skip_dim = -1) {
```

Comment:
A semi-random thought: how about a mask of unsplittable dimensions instead of just one?
```suggestion
                int min_sz = (16 << 10), uint64_t skip_dim_mask = 0) {
```

Reply:
done